### PR TITLE
fix: added confirmation dialog box for forgot pin

### LINKED ISSUE #1057: Add a confirmation dialog in forgot pin.
**Summary:**

When user clicks on forgot pin, he is directed to login page.

**Expected behavior:**

He should get a confirmation dialog.
https://github.com/openMF/mobile-wallet/issues/327

### PR DESCRIPTION
**Pull request by rchtgpt**
Fixes #1057 

GIF:
ezgif com-video-to-gif (8)

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by rchtgpt** [comment]
@miplodder @epaul hey, can you please review the task? thanks

File: app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java [hunk 1]
```diff
@@ -51,10 +55,24 @@ public void startNextActivity() {
 
     @Override
     public void startLoginActivity() {
-        Intent i = new Intent(PassCodeActivity.this, LoginActivity.class);
-        i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
-        startActivity(i);
-        finish();
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage("Are you sure you want to login using Password?");
```
Comment (laxyapahuja) [inline app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java]: Hey, shouldn't this not be hardcoded?

Comment (rchtgpt) [reply app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java]: Fixed it. Thanks for pointing out 😃

File: app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java [hunk 2]
```diff
@@ -51,10 +55,24 @@ public void startNextActivity() {
...
+                i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+                startActivity(i);
+                finish();
+            }
+        })
+                .setNegativeButton("No", null);
+        AlertDialog alert = builder.create();
+        alert.show();
+        Button nbutton = alert.getButton(DialogInterface.BUTTON_NEGATIVE);
+        nbutton.setTextColor(Color.BLUE);
```
Comment (elirehema) [inline app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java]: Hi @rachittt can you change `Color.BLUE` to use the same blue color used in application resources color `primary_dark ` both in `nbutton` and `pbutton`

**Comment by rachittt** [comment]
sure

File: app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java (hunk 2 above)
Comment (rchtgpt) [reply app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java]: sure

**Comment by ePaul** [comment]
@rachittt I think you pinged the wrong @ePaul, but it looks like @elirehema already found it.

**Comment by rchtgpt** [comment]
> @rachittt I think you pinged the wrong @ePaul, but it looks like @elirehema already found it.

Sorry, didn't mean to disturb you 😄

**Comment by luckyman20** [comment]
@rachittt Use `MaterialDialog` class in utils which we have in this project.

**Comment by rchtgpt** [comment]
@luckyman20 hey, made the requested changes. Kindly check

File: app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java [hunk 3]
```diff
@@ -51,10 +53,29 @@ public void startNextActivity() {
...
-        startActivity(i);
-        finish();
+        new MaterialDialog.Builder().init(PassCodeActivity.this)
+                .setCancelable(false)
+                .setMessage(R.string.login_using_password_confirmation)
+                .setPositiveButton(getString(R.string.logout),
+                        new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int which) {
+                                Intent i = new Intent(PassCodeActivity.this, LoginActivity.class);
```
Comment (luckyman20) [inline app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java]:
```suggestion
                                Intent intent = new Intent(PassCodeActivity.this, LoginActivity.class);
```

**Review by luckyman20 — CHANGES_REQUESTED** [review]

**Comment by rchtgpt** [comment]
@luckyman20 updated :)

**Review by luckyman20 — APPROVED** [review]
Looks good!

**Pull request merged**